### ISSUE TITLE
Set padded RTG in training data to be true RTG until masking is implemented correctly.

### ISSUE THREAD

**Issue by jbloomAus** [jbloomAus/DecisionTransformerInterpretability#43]
We aren't currently masking padding tokens which I'm not sure how much of a big deal it is. I'll reach out to ask someone for feedback, but in the meantime, I'm going to set the RTG in the padded training data to the true RTG because the padding RTG is 0 which I think might have confused the model.

This is hopefully going to be the different between a memory task solving DT completely solving the task and it's current status of solving it but not  using RTG to modulate behavior for the balls (it works if the key is the goal strangely). 

- [x] update offline training data to padd with true RTG 
- [x] message someone about how essential padding is (I'm not sure why it isn't already implemented on Hooked Transformer class, maybe it is and I missed it)

**Comment by jbloomAus** [comment]
I've done this which I think was helpful/good, and added tests for it. However, experimentally the results are still not dramatically better. 

Possible explanations:
- Eval is broken (seems unlikely since streamlit app also shows same issues and I've recently looked through both carefully)
- Training data sucks in some way. Specifically, if the 0 reward trajectories are either too variable  such that the agent doesn't unclear the non-RTG considering heuristics, or biased towards only one class, then the model may be overfitting on those (ie: a high variance issue). Part of the issue is of course the relationship between next token prediction and RL task performance. It's also worth noting the PPO model seems to have learnt to do random actions with no consequence like drop and done to increase entropy which don't contribute to solving the problem. It's almost certain our model is putting effort into doing this well which may be a form of overfitting depending on how present it is in the training set. I'll made a card for researching this later. 

Solutions to this probably look like one of:
    - Generate more training data
    - Generate better training data: Sample differently in rollouts? Sample from different quality agents? Add in some random sampling to go with the agent? 

The easiest things to try are:
1. Sample more.
2. Sample differently in the rollouts (deterministically might work given our weight regularization). 

Harder things to try are:
1. Merging rollouts from multiple agents.
2. Retraining PPO model to have less of the entropy maximizing behaviors.
